src/Bicep.Local.Deploy/LocalDeploymentSettings.cs

```diff
@@ -0,0 +1,262 @@
+
+    public int ResourceGroupConcurrentRequestLimit { get; set; } = 20;
+
+    public int ExportTemplateResourcesLimit { get; set; } = 200;
+
+    public bool ExportTemplateAsyncEnabled { get; set; } = true;
+
+    public int ExportTemplateSynchronousResourcesLimit { get; set; } = 2;
+
+    public TimeSpan ExportTemplateSynchronousWaitDuration { get; set; } = TimeSpan.FromSeconds(5);
+
+    public TimeSpan ExportTemplateJobTimeout { get; set; } = TimeSpan.FromMinutes(15);
+
+    public TimeSpan AsyncOperationRetryAfterInterval { get; set; } = TimeSpan.FromSeconds(15);
+
+    public bool UseLocationBasedAsyncURI { get; set; } = true;
+
+    public double WhatIfLiveTrafficValidationPercentage { get; set; } = 0.2;
+
+    public string[] WhatIfLiveTrafficValidationApplicationWhitelist { get; set; } = new string[] { };
+
+    public string[] WhatIfLiveTrafficValidationTenantBlacklist { get; set; } = new string[] { };
+
+    public string[] WhatIfLiveTrafficValidationSubscriptionBlacklist { get; set; } = new string[] { };
+
+    public string DefaultFrontdoorEndpoint { get; set; } = "localhost";
+
+    public int SyncPreflightResourceGroupLimit { get; set; } = 10;
+
+    public int SyncPreflightNestedDeploymentExpansionLimit { get; set; } = 50;
+
+    public TimeSpan SyncPreflightNestedDeploymentExpansionTimeout { get; set; } = TimeSpan.FromSeconds(30);
+
+    public int AsyncPreflightResourceGroupLimit { get; set; } = 800;
+
+    public int AsyncPreflightNestedDeploymentExpansionLimit { get; set; } = 500;
+
+    public TimeSpan AsyncPreflightNestedDeploymentExpansionTimeout { get; set; } = TimeSpan.FromMinutes(5);
+
+    public int ResourceValidationRequestSynchronousLimit { get; set; } = 10;
+
+    public KeyValuePair<string, string>[] BlacklistedDeploymentParameters { get; set; } = new KeyValuePair<string, string>[] { };
+
+    public TimeSpan DeploymentJobTimeout { get; set; } = TimeSpan.FromSeconds(60);
+
+    public Uri ExtensibilityHostUri { get; set; } = new Uri("https://example.com");
+
+    public IReadOnlyDictionary<string, Uri> ExtensibilityHostUriOverridesByTenantId { get; set; } = new Dictionary<string, Uri>();
+
+    public bool ShadowTestingModeEnabled { get; set; } = false;
+
+    public double ShadowTestingTrafficValidationPercentage { get; set; } = 0.2;
+
+    public string? ShadowTestingEndpoint { get; set; }
+
+    public TimeSpan PreflightJobMinimumRetryInterval { get; set; } = TimeSpan.FromSeconds(value: 10);
+
+    public TimeSpan PreflightJobMaximumRetryInterval { get; set; } = TimeSpan.FromMinutes(value: 2);
+
+    public string[] PreviewFunctionalityPermittedTenants { get; set; } = new string[] { };
+
+    public string[] PreviewFunctionalityDisabledFeatures { get; set; } = new string[] { };
+
+    public string[] PreviewFunctionalityExternalPermittedTenants { get; set; } = new string[] { };
+
+    public string[] PreviewFunctionalityPublicFeatures { get; set; } = new string[] {
```

Comment:
I'm probably a little confused but where does this get checked compared to the ServiceDirectory.json? Or is it not at all?

Reply:
It's not so relevant for this particular feature, but I'm forced to define a value for it.